### ISSUE TITLE
ci: Timeline to drop Mac High Sierra + Mac Python 2 support

### ISSUE THREAD

**Issue by EricCousineau-TRI** [RobotLocomotion/drake#11692]
To reduce number of builds, we may want to turn down on Mac by turning of High Sierra.

~Any hard votes against doing this by September, 2019? (in preparation for Mac Catalina)~
Plan is to turn this off once Catalina is out, possibly once we start adding beta CI configurations (per Jermey's comment below).

Relates #11610 

\cc @jwnimmer-tri @RussTedrake @soonho-tri

**Comment by jwnimmer-tri** [comment]
In my opinion we should turn off High Sierra CI configurations exactly when we begin adding Catalina CI configurations, per our de-facto {N, N-1} policy for macOS.

**Comment by soonho-tri** [comment]
- Apple says that "macOS Catalina (10.15) will be released in the fall of 2019" without further details.
 - I guess it will take some time for our CI infrastructures to be ready to run 10.15. (as it happened last year). So I expect that we will have 2 - 3 months that we will only run 10.14 in CI if we drop 10.13.

So, if we want to stick to the N/N-1 policy for macOS, we need to wait until 10.15 is available in CI. I'm fine if we want to change the policy temporarily just to supporting 10.14.

**Comment by jwnimmer-tri** [comment]
Let me clarify my position in light of that: we should turn off High Sierra CI configurations exactly when Catalina is released, or we begin adding Catalina beta CI configurations, whichever comes first.

**Comment by jamiesnape** [comment]
My guess would be that Catalina builds will be up fairly quickly (within a month of release). Mojave was unusual since the required filesystem type changed.

**Comment by RussTedrake** [comment]
fwiw -- i have no attachment to high sierra.  all the mac use cases I'm aware of are on Mojave.

**Comment by EricCousineau-TRI** [comment]
Relates #12003; from that discussion, and the Slack poll, seems like no one really needs Python 2 on Mac, and Homebrew's dropping of Python 2 support means that we may want to drop this as well.

Updated the checkbox in #10606 to xref this issue.

**Comment by jamiesnape** [comment]
Note that we are waiting on `clp`, `diffstat`, `doxygen`, and `llvm@6` dependencies to be built for Catalina for unprovisioned builds to be switched on.

**Comment by EricCousineau-TRI** [comment]
Marking this as resolved for now; will create separate issue about Catalina.
